lib/jekyll/document.rb

```diff
@@ -51,24 +52,28 @@ def data
     #
     # Returns the merged data.
     def merge_data!(other, source: "YAML front matter")
-      if other.key?('categories') && !other['categories'].nil?
-        if other['categories'].is_a?(String)
-          other['categories'] = other['categories'].split(" ").map(&:strip)
-        end
-        other['categories'] = (data['categories'] || []) | other['categories']
-      end
+      merge_categories!(other)
       Utils.deep_merge_hashes!(data, other)
-      if data.key?('date') && !data['date'].is_a?(Time)
-        data['date'] = Utils.parse_date(
-          data['date'].to_s,
+      if data.key?("date") && !data["date"].is_a?(Time)
+        data["date"] = Utils.parse_date(
+          data["date"].to_s,
           "Document '#{relative_path}' does not have a valid date in the #{source}."
         )
       end
       data
     end
 
+    def merge_categories!(other)
```

Comment:
At the bottom of the class and each one gets an explicit private.  Rubocop doesn't like that because they consider it useless but I consider it good form to be explicit because one could easily miss a private 20 methods away when quickly viewing a file.